cleanrl/rnd_ppo.py

```diff
@@ -0,0 +1,1079 @@
+    if frame_stack:
+        env = FrameStack(env, 4)
+    return env
+
+
+class ImageToPyTorch(gym.ObservationWrapper):
+    """
+    Image shape to channels x weight x height
+    """
+
+    def __init__(self, env):
+        super(ImageToPyTorch, self).__init__(env)
+        old_shape = self.observation_space.shape
+        self.observation_space = gym.spaces.Box(
+            low=0,
+            high=255,
+            shape=(old_shape[-1], old_shape[0], old_shape[1]),
+            dtype=np.uint8,
+        )
+
+    def observation(self, observation):
+        return np.transpose(observation, axes=(2, 0, 1))
+
+def wrap_pytorch(env):
+    return ImageToPyTorch(env)
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.nn.functional as F
+from torch.distributions.categorical import Categorical
+from torch.utils.tensorboard import SummaryWriter
+
+import argparse
+from distutils.util import strtobool
+import numpy as np
+import gym
+import vizdoomgym
+from gym.wrappers import TimeLimit, Monitor
+import pybullet_envs
+from gym.spaces import Discrete, Box, MultiBinary, MultiDiscrete, Space
+import time
+import random
+import os
+from stable_baselines3.common.vec_env import DummyVecEnv, SubprocVecEnv, VecEnvWrapper
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='PPO agent')
+    # Common arguments
+    parser.add_argument('--exp-name', type=str, default=os.path.basename(__file__).rstrip(".py"),
+                        help='the name of this experiment')
+    parser.add_argument('--gym-id', type=str, default="BreakoutNoFrameskip-v4",
+                        help='the id of the gym environment')
+    parser.add_argument('--learning-rate', type=float, default=2.5e-4,
+                        help='the learning rate of the optimizer')
+    parser.add_argument('--seed', type=int, default=1,
+                        help='seed of the experiment')
+    parser.add_argument('--total-timesteps', type=int, default=10000000,
+                        help='total timesteps of the experiments')
+    parser.add_argument('--torch-deterministic', type=lambda x:bool(strtobool(x)), default=True, nargs='?', const=True,
+                        help='if toggled, `torch.backends.cudnn.deterministic=False`')
+    parser.add_argument('--cuda', type=lambda x:bool(strtobool(x)), default=True, nargs='?', const=True,
+                        help='if toggled, cuda will not be enabled by default')
+    parser.add_argument('--prod-mode', type=lambda x:bool(strtobool(x)), default=False, nargs='?', const=True,
+                        help='run the script in production mode and use wandb to log outputs')
+    parser.add_argument('--capture-video', type=lambda x:bool(strtobool(x)), default=False, nargs='?', const=True,
+                        help='weather to capture videos of the agent performances (check out `videos` folder)')
+    parser.add_argument('--video-interval', type=int, default=50,
```

Comment:
In our experiments. From the 7th(343) video being recorded to 8th(512) video being recorded. It took almost a day. And it will not take too much space as each video is about 50k.